scripts/extensions/pool_update.apply

```diff
@@ -115,9 +115,13 @@ if __name__ == '__main__':
             print(success_message())
         except Exception as e:
             print(failure_message(PATCH_APPLY_FAILED,
-                                  ['Apply failed: %s' % e]))
+                                  ['Failed to apply the update: %s' % e]))
         finally:
-            session.xenapi.pool_update.detach(update)
+            try:
+                session.xenapi.pool_update.detach(update)
+            except Exception as e:
+                print(failure_message(PATCH_APPLY_FAILED,
```

Comment:
`detach` failure can occur if the update VDI is destroyed before prechek/apply is run.